### ISSUE TITLE
Merge neighbourhood processing interface into fewer CLIs

### ISSUE THREAD

**Issue by cgsandford** [metoppv/improver#481]
As an IMPROVER developer I want a clear neighbourhood processing interface so that it is easy to work with.

This ticket is to merge the "nbhood", "nbhood-iterate-with-mask" and "nbhood-land-and-sea" CLIs into a single "nbhood" CLI, with the following parameters and defaults:

- New option "--kernel" to specify circle, circle_weighted, square or square_recursive filtering
- Make square_recursive the default for this CLI, with default alphas values

There is a significant amount of code in some of these CLIs, and follow on work may be required to move this into the IMPROVER libraries.

Related issues: #480 (prerequisite), #352 (example behaviour of new CLI)

Acceptance criteria:
 * A single CLI that preserves existing functionality (after #480), with the additional parameter and default defined above
 * Rationalise bats test and known good output for these CLIs, removing duplicates and reflecting the new functionality and defaults
 * Identify follow on work to move code out of this CLI and into IMPROVER plugins or utilities
 * Fixing the suite to use the new CLI is out of scope and will be covered by IMPRO-589

**Comment by fionaRust** [comment]
Superseded by https://github.com/metoppv/mo-blue-team/issues/157
